simulation/halsim_print/src/halsim_print/cpp/PrintPWM.cpp

```diff
@@ -0,0 +1,29 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "PrintPWM.h"
+
+#include <iostream>
+
+#include "MockData/HAL_Value.h"
+#include "MockData/NotifyListener.h"
+#include "MockData/PWMData.h"
+
+static void pwm_callback(const char* name, void* param,
```

Comment:
Use camel case function name instead of snake case.

Reply:
I guess I did read this wrong:
  https://github.com/wpilibsuite/styleguide/blob/master/cppguide.html#L801
in that this is a small function; it's just not inline.  But I personally find it valuable to make a clear distinction between class methods and static functions.  In fact, you see that pattern throughout the existing code (i.e. HALSIM_xxx, HAL_yyy).  And since it was static and local to the file, lowercase felt appropriate to me.  (i.e. if I was to export that function I would have felt PrintPWM_Callback was an appropriate name).  I will change this if it is required for it to be committed, but I would prefer to leave it as is.

Reply:
`HAL_` is used as a C programmer's namespace. That's the only place you will see underscores used besides legacy macro code in WPIErrors.h and the `m_` in member variables. For function naming purposes, you basically ignore the underscore prefix. The HAL doesn't seem to be consistent on capitalizing the first letter on non-member functions, but it all uses camel case or mixed case. Yet another thing I should ask about and go clean up...